src/graia/ariadne/message/parser/alconna.py

```diff
@@ -0,0 +1,469 @@
+                                _text_static[""] += 1
+                except (IndexError, KeyError):
+                    break
+        else:
+            _head = True
+        try:
+            # 如果没写options并且marg不是str的话，匹配完命令头后是进不去上面的代码的，这里单独拿一段出来
+            may_element_index = self.result.raw_texts[self.result.current_index][1] + 1
+            if (
+                self.result.elements
+                and type(self.result.elements[may_element_index]) is self._params[0]
+            ):
+                self.result.results["main_argument"] = self.result.elements[
+                    may_element_index
+                ]
+                del self.result.elements[may_element_index]
+        except (IndexError, KeyError):
+            pass
+
+        if (
+            _head
+            and len(self.result.elements) == 0
+            and all([t[0] == "" for t in self.result.raw_texts])
+            and (
+                not self.result.need_marg
+                or (self.result.need_marg and "main_argument" in self.result.results)
+            )
+        ):
+            self.result.matched = True
+            self.result.analysis_result()
+        else:
+            self.result.results.clear()
+        return self.result
+
+    def _analysis_option(self, param, text, rest_text, option_dict):
+        opt = param["name"]
+        arg = param["args"]
+        sep = param["separator"]
+        name, may_arg = self.split_once(text, sep)
+        if sep == self.separator:  # 在sep等于separator的情况下name是被提前切出来的
+            name = text
+        if not re.match("^" + opt + "$", name):  # 先匹配选项名称
+            return False
+        self.result.raw_texts[self.result.current_index][0] = rest_text
+        if arg == {}:
+            option_dict[text] = text
+            return True
+        for k, v in arg.items():
+            if isinstance(v, str):
+                if sep == self.separator:
+                    may_arg, rest_text = self.result.split_by(sep)
+                if not re.match("^" + v + "$", may_arg):
+                    return False
+                if name not in option_dict:
+                    option_dict[name] = {k: may_arg}
+                else:
+                    option_dict[name][k] = may_arg
+                if sep == self.separator:
+                    self.result.raw_texts[self.result.current_index][0] = rest_text
+                return True
+            else:
+                may_element_index = (
+                    self.result.raw_texts[self.result.current_index][1] + 1
+                )
+                if type(self.result.elements[may_element_index]) is not v:
+                    return False
+                if name not in option_dict:
+                    option_dict[name] = {k: self.result.elements[may_element_index]}
+                option_dict[name][k] = self.result.elements[may_element_index]
+                del self.result.elements[may_element_index]
+                return True
```

Comment:
there is a little bug...
update with lasted Alconna  [`c5f2288`](https://github.com/RF-Tar-Railt/Cesloi-Alconna/commit/c5f2288718ad6b18aeaec4c24ef05d855ddf74f8#diff-bb012d64de22456b65147c1230ae7b5509d31f4c01186952b9c780e8ff961bb5)

Reply:
> 
lasted Alconna [`6e619dd`](https://github.com/RF-Tar-Railt/Cesloi-Alconna/commit/6e619dda14a6e6d40aed782edab891055f239236#diff-bb012d64de22456b65147c1230ae7b5509d31f4c01186952b9c780e8ff961bb5)